api/desecapi/permissions.py

```diff
@@ -9,3 +9,12 @@ class IsOwner(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         return obj.owner == request.user
 
+
+class IsDomainOwner(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of a domain to view or edit an object owned by that domain.
```

Comment:
I don't think this is misleading.

1) The comment doesn't say that the permission is active without being called. It is still a "permission" in Django speech. Consequently, the parent class is a BasePermission (and not a "CallablePermissionCheck" or something).

2) This follows the same language as the other permission in that file that you implemented ;-)